### ISSUE TITLE
haproxy 2.4.25 converter map doesn't set default value for some kinds of empty strings

### ISSUE THREAD

**Issue by morgana2313** [haproxy/haproxy#2426]
### Detailed Description of the Problem

Haproxy version `2.4.25-1ppa1~bionic 2023/12/16` doesn't set the default value for a map if the input is empty:

```
    http-request        set-var(txn.foo) path,word(1,"/") # get first word from path
    http-request        set-var(txn.bar) var(txn.foo), map(foobar.map,default_value) # foobar.map is an empty file.

    http-response       set-header foobar1 "foo:'%[var(txn.foo)]' bar:'%[var(txn.bar)]'"

    http-request        set-var(txn.bar) str(default_value) if { var(txn.bar) -m str }

    http-response       set-header foobar2 "foo:'%[var(txn.foo)]' bar:'%[var(txn.bar)]'"
```

`http-request        set-var(txn.foo) str()` is a different kind of empty and doesn't manifest this problem. The if command also doesn't match this kind of emptyness?

### Expected Behavior

I expect the default value to be set, just as it was in version `2.4.22-1ppa1~bionic 2023/02/16` and before.

### Steps to Reproduce the Behavior

I think this is wrong:

```
$ curl --head http://hak5.sensornw.net/
HTTP/1.1 200 OK
[...]
foobar1: foo:'' bar:''
foobar2: foo:'' bar:''
```

I think this is rignt:
```
curl --head http://hak5.sensornw.net/X # <======== note the X!
HTTP/1.1 404 Not Found
[...]
foobar1: foo:'X' bar:'default_value'
foobar2: foo:'X' bar:'default_value'
```

### Do you have any idea what may have caused this?

Upgrade to version `2.4.25-1ppa1~bionic 2023/12/16` 

### What is your configuration?

```haproxy
Linux Ubuntu 18.04.6
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.4.25-1ppa1~bionic 2023/12/16 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2026.
Known bugs: http://www.haproxy.org/bugs/bugs-2.4.25.html
Running on: Linux 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -O2 -fdebug-prefix-map=/build/haproxy-2sEpO3/haproxy-2.4.25=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_SYSTEMD=1 USE_PROMEX=1
  DEBUG   =

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL +EPOLL -EVPORTS +FUTEX +GETADDRINFO -KQUEUE +LIBCRYPT +LINUX_SPLICE +LINUX_TPROXY +LUA -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OT -PCRE +PCRE2 +PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PRIVATE_CACHE -PROCCTL +PROMEX -PTHREAD_PSHARED -QUIC +RT +SLZ -STATIC_PCRE -STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL -ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=40).
Built with OpenSSL version : OpenSSL 1.1.1  11 Sep 2018
Running on OpenSSL version : OpenSSL 1.1.1  11 Sep 2018
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.31 2018-02-12
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 7.5.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
	[SPOE] spoe
	[CACHE] cache
	[FCGI] fcgi-app
	[COMP] compression
	[TRACE] trace
```

**Comment by Darlelet** [comment]
I didn't test, but this could be a side effect of 1dcc6a8a969893057e56d4eba089a538619c72c2, since now in your case (when you don't provide a path), `txn.foo` probably won't exist anymore with the mentioned patch.

**Comment by Darlelet** [comment]
For reference, 1dcc6a8a969893057e56d4eba089a538619c72c2 is shipped with 2.4.25

**Comment by Darlelet** [comment]
Indeed, I can confirm this is due to 1dcc6a8a969893057e56d4eba089a538619c72c2 after testing.

Too bad this causes a regression on your side, but I think it is intended behavior since the `field` converter already behaves like this for a while now and the fix was explicitly flagged as a bug and set for backports.
